frontend/apps/reader/modules/readerfooter.lua

```diff
@@ -511,46 +525,181 @@ function ReaderFooter:addToMainMenu(menu_items)
+                            return self.settings.progress_perc_format == "1"
+                        end,
+                        callback = function()
+                            self.settings.progress_perc_format = "1"
+                            self:updateFooter()
+                            UIManager:setDirty(nil, "ui")
+                        end,
+                    },
+                    {
+                        text = T(_("2 digits after decimal point (%1)"), self:progressPercentage(2)),
+                        checked_func = function()
+                            return self.settings.progress_perc_format == "2"
+                        end,
+                        callback = function()
+                            self.settings.progress_perc_format = "2"
+                            self:updateFooter()
+                            UIManager:setDirty(nil, "ui")
+                        end,
+                    },
+                },
+            },
+            {
+                text = _("Date format"),
+                sub_item_table = {
+                    {
+                        text = _("24-hour"),
+                        checked_func = function()
+                            return self.settings.date_format == "24" or self.settings.date_format == nil
+                        end,
+                        callback = function()
+                            self.settings.date_format = "24"
+                            self:updateFooter()
+                            UIManager:setDirty(nil, "ui")
+                        end,
+                    },
+                    {
+                        text = _("12-hour"),
+                        checked_func = function()
+                            return self.settings.date_format == "12"
+                        end,
+                        callback = function()
+                            self.settings.date_format = "12"
+                            self:updateFooter()
+                            UIManager:setDirty(nil, "ui")
+                        end,
+                    },
+                }
+            },
+            {
+                text = _("Time format"),
+                sub_item_table = {
+                    {
+                        text = _("Classic"),
+                        checked_func = function()
+                            return self.settings.time_format == "classic" or self.settings.time_format == nil
+                        end,
+                        callback = function()
+                            self.settings.time_format = "classic"
+                            self:updateFooter()
+                            UIManager:setDirty(nil, "ui")
+                        end,
+                    },
+                    {
+                        text = _("Modern"),
+                        checked_func = function()
+                            return self.settings.time_format == "modern"
```

Comment:
Bit confusing `Date format` (above) vs `Time format` (here).
Your Date format (12/24) indeed sets the formating of the time.
Your Time format (classic/modern) actually set the style of **durations** (12:04 vs 12m4s).

Reply:
Good tip. Switched.